File: app/src/main/java/com/king/frame/utils/AppUtil.java
package com.king.frame.utils;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Build;

/**
 * @author king
 * @date 2017/3/6 16:33
 */

public class AppUtil {

    /**
     * 获取app版本
     * @param context
     * @return
     */
    public static int getAppVersionCode(Context context) {
        if (context != null) {
            PackageManager pm = context.getPackageManager();
            if (pm != null) {
                PackageInfo pi;
                try {
                    pi = pm.getPackageInfo(context.getPackageName(), 0);
                    if (pi != null) {
                        return pi.versionCode;
                    }
                } catch (PackageManager.NameNotFoundException e) {
                    e.printStackTrace();
                }
            }
        }
        return -1;
    }

    /**
     * 获取app版本名
     * @param context
     * @return
     */
    public static String getAppVersionName(Context context) {
        if (context != null) {
            PackageManager pm = context.getPackageManager();
            if (pm != null) {
                PackageInfo pi;
                try {
                    pi = pm.getPackageInfo(context.getPackageName(), 0);
                    if (pi != null) {
                        return pi.versionName;
                    }
                } catch (PackageManager.NameNotFoundException e) {
                    e.printStackTrace();
                }
            }
        }
        return null;
    }

    /**
     * 获取手机安卓版本代号
     * @return
     */
    public static int getAndroidVersion(){
        return Build.VERSION.SDK_INT;
    }

    /**
     * 获取手机机型
     * @return
     */
    public static String getMobileModel(){
        return android.os.Build.MODEL;
    }

    /**
     * 获取手机生产厂家
     * @return
     */
    public static String getMobileManufacturer(){
        return android.os.Build.MANUFACTURER;
    }
}

File: app/src/main/java/com/king/frame/utils/AppUtil.kt
package com.king.frame.utils

import android.content.Context
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.os.Build

/**
 *  App/系统相关工具类
 *
 * @author king
 * @date 2019-11-27 10:33
 */
object AppUtil {

    /**
     * 获取手机安卓版本代号
     * @return
     */
    val androidVersion: Int
        get() = Build.VERSION.SDK_INT

    /**
     * 获取手机机型
     * @return
     */
    val mobileModel: String
        get() = android.os.Build.MODEL

    /**
     * 获取手机生产厂家
     * @return
     */
    val mobileManufacturer: String
        get() = android.os.Build.MANUFACTURER

    /**
     * 获取app版本
     * @param context
     * @return
     */
    fun getAppVersionCode(context: Context?): Int {
        if (context != null) {
            val pm = context.packageManager
            if (pm != null) {
                val pi: PackageInfo?
                try {
                    pi = pm.getPackageInfo(context.packageName, 0)
                    if (pi != null) {
                        return pi.versionCode
                    }
                } catch (e: PackageManager.NameNotFoundException) {
                    e.printStackTrace()
                }

            }
        }
        return -1
    }

    /**
     * 获取app版本名
     * @param context
     * @return
     */
    fun getAppVersionName(context: Context?): String? {
        if (context != null) {
            val pm = context.packageManager
            if (pm != null) {
                val pi: PackageInfo?
                try {
                    pi = pm.getPackageInfo(context.packageName, 0)
                    if (pi != null) {
                        return pi.versionName
                    }
                } catch (e: PackageManager.NameNotFoundException) {
                    e.printStackTrace()
                }

            }
        }
        return null
    }
}
